src/components/Table/Table.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from 'react';
+import { Person } from '../../types';
+import { useParams } from 'react-router-dom';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+
+export const Table: React.FC = () => {
+  const { personSlug } = useParams();
+  const [requestState, setRequestState] = useState('loading');
+  const [peopleTable, setPeopleTable] = useState<Person[] | null>(null);
+
+  useEffect(() => {
+    fetch('https://mate-academy.github.io/react_people-table/api/people.json');
+    getPeople()
+      .then((data: Person[]) => {
+        if (data.length === 0) {
```

Comment:
Why are you checking this?
If the array is empty, `treatedData` will also be an empty array, the _.map_ method will not run on an empty array...